### DISCUSSION TITLE
Help with tweens

### DISCUSSION THREAD

**Discussion by EPiC-Inc** [vcync/modV#659] [Q&A] [answered]
How do I use the tween editor? I can't find anything about it on the wiki or the documentation.

**Comment by 2xAA** [comment, marked as answer]
Hey @EPiC-Inc, so the tween editor really needs a redesign. It's a real pain to use and I think has an inherent bug which adds half the total duration to itself...

_But_, I'll walk you through it as it stands.

1. First of all, have a Module ready and select one of the props so it shows up in the Input Config panel.
    I've selected Size on the Text Module here.

    ![Screenshot 2021-11-25 at 11 14 15](https://user-images.githubusercontent.com/554219/143431670-9a5625ab-2710-423a-9a5f-d7e7adf19a23.png)

2. Play around with the Module prop and find a few values you'd like to animate between.
    I found 100 and 200 for the Text Module's size worked well for me.
    
    The format for the Data is an array of arrays, e.g. `[[100], [200]]`.
    Type this into the Data input and press tab or click away from the input to make it lose the focus ring - your Tween will update.
    
    ![Screenshot 2021-11-25 at 11 14 41](https://user-images.githubusercontent.com/554219/143432364-850a7831-895f-48e3-9161-40f761098385.png)

#### Data
To animate between two values, use: `[[100], [200]]`
To animate between three values, use `[[100], [200], [300]]`
etc.

You must press tab or lose focus on the text input to update the Tween with the new data.

#### Easing
See an example of all easings here: https://easings.net/

#### Duration
The duration of your tween in milliseconds, either the duration of each step or the total tween time - dictated by "Duration as total time"

#### Use BPM
If selected, the duration will be the BPM divided by the BPM Divison

#### BPM Division
See Use BPM

#### Duration as total time
See Duration

#### Steps
If set to 0, the tween will be keyframed over duration divided by the current FPS.
If larger than 0, the keyframes will be evenly spread over the duration time divided by the steps. e.g. if set to 4 and your duration is 1000, you will have 4 keyframes which update every 250 milliseconds.

**Comment by EPiC-Inc** [comment]
Thank u! Idk why I didn't see this [tho to be fair I haven't used GH for a hot second].
As it stands, it still works for me :]
